task/github-app-token/0.2/github-app-token.yaml

```diff
@@ -0,0 +1,161 @@
+# TODO: Submit this in catalog upstream
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-app-token
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: github
+    tekton.dev/displayName: "github app token"
+spec:
+  description: >-
+    Retrive a user token from a GitHub application
+
+    This task will get a user token for an installation_id for a GitHub application.
+    This could be then reuse to do user operations.
+
+  workspaces:
+    - name: secrets
+
+  results:
+    - name: token
+      description: The user token to use.
+
+  params:
+    - name: installation_id
+      description: The installation id for the GitHub application to request a token for.
+      type: string
+
+    - name: application_id
+      description: The application id for the GitHub application to request a token for.
+      type: string
+      default: ""
+
+    - name: private_key_path
+      description: The key path inside the secret workspace
+      type: string
+      default: "private.key"
+
+    - name: application_id_path
+      description: The path for the application id inside the secret workspace
+      type: string
+      default: "application_id"
+
+    - name: token_expiration_minutes
+      description: Token expiration time in minutes
+      type: string
+      default: "10"
+
+  steps:
+    - name: get-token
+      image: quay.io/chmouel/github-app-token@sha256:bc45937ae588df876555ebb56c36350ed74592c7f55f2df255cabef39c926a88
+      env:
+        - name: GITHUBAPP_KEY_PATH
+          value: $(workspaces.secrets.path)/$(inputs.params.private_key_path)
+        - name: GITHUBAPP_APP_ID_PATH
+          value: $(workspaces.secrets.path)/$(inputs.params.application_id_path)
+        - name: GITHUBAPP_APP_ID
+          value: $(inputs.params.application_id)
```

Comment:
No need for the `inputs.` prefix in v1beta1.  Same for all other param variables.

Reply:
I always forget, old timer curse 😅